components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.email/src/main/java/org/wso2/carbon/event/output/adapter/email/EmailEventAdapter.java

```diff
@@ -14,57 +15,76 @@
-import org.apache.axis2.transport.base.BaseConstants;
 import org.apache.axis2.transport.mail.MailConstants;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+
 import org.wso2.carbon.event.output.adapter.core.OutputEventAdapter;
 import org.wso2.carbon.event.output.adapter.core.OutputEventAdapterConfiguration;
+import org.wso2.carbon.event.output.adapter.core.exception.ConnectionUnavailableException;
 import org.wso2.carbon.event.output.adapter.core.exception.OutputEventAdapterException;
 import org.wso2.carbon.event.output.adapter.core.exception.TestConnectionNotSupportedException;
 import org.wso2.carbon.event.output.adapter.email.internal.util.EmailEventAdapterConstants;
-import org.wso2.carbon.event.output.adapter.email.internal.ds.EmailEventAdapterServiceValueHolder;
 
-import java.util.HashMap;
+import javax.mail.*;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import java.util.Date;
 import java.util.Map;
+import java.util.Properties;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
+
+/**
+ * The Email event adapter sends mail using an SMTP server configuration defined
+ * in output-event-adapters.xml email adapter sender definition.
+ */
+
 public class EmailEventAdapter implements OutputEventAdapter {
 
     private static final Log log = LogFactory.getLog(EmailEventAdapter.class);
     private static ThreadPoolExecutor threadPoolExecutor;
+    private static Session session;
     private OutputEventAdapterConfiguration eventAdapterConfiguration;
     private Map<String, String> globalProperties;
 
+
+    /**
+     * Default from address for outgoing messages.
+     */
+    private InternetAddress smtpFromAddress = null;
+
+
     public EmailEventAdapter(OutputEventAdapterConfiguration eventAdapterConfiguration,
                              Map<String, String> globalProperties) {
         this.eventAdapterConfiguration = eventAdapterConfiguration;
         this.globalProperties = globalProperties;
+
     }
 
+    /**
+     * Initialize the thread pool to send emails.
+     *
+     * @throws OutputEventAdapterException on error.
+     */
+
     @Override
     public void init() throws OutputEventAdapterException {
 
-        //ThreadPoolExecutor will be assigned  if it is null
+        //ThreadPoolExecutor will be assigned  if it is null.
         if (threadPoolExecutor == null) {
             int minThread;
             int maxThread;
             long defaultKeepAliveTime;
 
+
             //If global properties are available those will be assigned else constant values will be assigned
             if (globalProperties.get(EmailEventAdapterConstants.MIN_THREAD_NAME) != null) {
                 minThread = Integer.parseInt(globalProperties.get(EmailEventAdapterConstants.MIN_THREAD_NAME));
```

Comment:
What happens if the configured value is not an integer ?. Even if this is not done by you, Better if you can fix it. I think there should be a default value and, you should check whether the configured value is an int. If it is not an int you can assign default value. (Hope there is a string utils function)